### PR TITLE
Replace Gas, GasCost and ProgramCounter by u64.

### LINKED ISSUE #1404: Replace `ProgramCounter` `Gas` and `GasCost` with simple u64
- wrapper types `ProgramCounter` `Gas` and `GasCost` add verbosity to the code but don't provide many benefits;
- We would avoid having to do things like `Gas(0)` instead of just `0`.
 
Originally posted by @ed255 in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1403#discussion_r1190957200_

### PR DESCRIPTION
**Pull request by rrtoledo**
### Description
This is a very small MRs removing some overhead. It replaces the types/structures for Gas, GasCost and ProgramCounter by u64.

### Issue Link
Issue #1404 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Rationale
Keeping the existing code but removing the structures.

### How Has This Been Tested?
No tests were added.

### CODE REVIEW

**Review by adria0 — APPROVED** [review]
LGTM!  I love this clean-ups! :)

File: eth-types/src/evm_types.rs [hunk 1]
```diff
@@ -14,171 +14,113 @@ pub use opcode_ids::OpcodeId;
...
-impl fmt::Display for GasCost {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_fmt(format_args!("{}", self.0))
-    }
-}
+// impl fmt::Display for GasCost {
+//     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+//         f.write_fmt(format_args!("{}", self.0))
+//     }
+// }
```
Comment (ChihChengLiang) [inline eth-types/src/evm_types.rs:30]: Can we remove the comment off code?

File: eth-types/src/evm_types.rs [hunk 2]
```diff
@@ -14,171 +14,113 @@ pub use opcode_ids::OpcodeId;
...
 /// Quotient for max refund of gas used
 pub const MAX_REFUND_QUOTIENT_OF_GAS_USED: usize = 5;
 /// Gas stipend when CALL or CALLCODE is attached with value.
 pub const GAS_STIPEND_CALL_WITH_VALUE: u64 = 2300;
 
 /// Defines the gas consumption.
-#[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Default)]
-pub struct GasCost(pub u64);
+// #[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Default)]
+pub struct GasCost;
```
Comment (ChihChengLiang) [inline eth-types/src/evm_types.rs:24]: It seems these two are no longer needed.
```suggestion
```

File: eth-types/src/evm_types.rs [hunk 3]
```diff
@@ -14,171 +14,113 @@ pub use opcode_ids::OpcodeId;
...
-impl fmt::Debug for GasCost {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_fmt(format_args!("{}", self.0))
-    }
-}
+// impl fmt::Debug for GasCost {
+//     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+//         f.write_fmt(format_args!("{}", self.0))
+//     }
+// }
```
Comment (ChihChengLiang) [inline eth-types/src/evm_types.rs:36]: Also here
```suggestion
```

File: eth-types/src/evm_types.rs [hunk 4]
```diff
@@ -14,171 +14,113 @@ pub use opcode_ids::OpcodeId;
...
+//     fn from(cost: u8) -> Self {
+//         GasCost(cost as u64)
+//     }
+// }
+
+// impl From<u64> for GasCost {
+//     fn from(cost: u64) -> Self {
+//         GasCost(cost)
+//     }
+// }
```
Comment (ChihChengLiang) [inline eth-types/src/evm_types.rs:126]:
```suggestion
```

**Review by ChihChengLiang — COMMENTED** [review]
Nice cleanup!
I left some comments on the leftover code.

File: eth-types/src/evm_types.rs (hunk 1 above)
Comment (rrtoledo) [reply eth-types/src/evm_types.rs:30]: Sorry! I forgot to remove them, of course!

File: eth-types/src/evm_types.rs (hunk 4 above)
Comment (rrtoledo) [reply eth-types/src/evm_types.rs]: Updated

File: eth-types/src/evm_types.rs (hunk 2 above)
Comment (rrtoledo) [reply eth-types/src/evm_types.rs:24]: We would still need the struct though (to implement it), am I wrong?

Comment (rrtoledo) [reply eth-types/src/evm_types.rs]: The comment was removed.

File: eth-types/src/evm_types.rs (hunk 3 above)
Comment (rrtoledo) [reply eth-types/src/evm_types.rs]: Updated

File: eth-types/src/evm_types.rs (hunk 2 above)
Comment (ChihChengLiang) [reply eth-types/src/evm_types.rs]: Yes, you are right. We still need to keep the GasCost struct.
I wasn't aware the constants are defined within the struct.

**Review by ChihChengLiang — APPROVED** [review]
LGTM! Nice clean up! Now we can use plain u64 for gas and program counters.
Feel free to merge.

**Pull request merged**